ENIGMAsystem/SHELL/Graphics_Systems/General/GSstdraw.cpp

```diff
@@ -415,11 +416,77 @@ void draw_triangle_color(gs_scalar x1, gs_scalar y1,gs_scalar x2, gs_scalar y2,
 }
 
 void draw_roundrect(gs_scalar x1, gs_scalar y1,gs_scalar x2, gs_scalar y2, float rad, bool outline) {
-	//TODO: Needs written to use circle precision for the corners
+	if (x1>x2) {
+		float t=x2;
+		x2=x1;
+		x1=t;
+    }
+  if (y1>y2) {
+    float t=y2;
+		y2=y1;
+		y1=t;
+	}
+
+	double pr = 2 * M_PI / enigma::circleprecision;
+	if (outline) {
```

Comment:
Not terribly important, but I'd recommend `draw_primitive_begin(outline ? pr_linestrip : pr_trianglefan)`.